LLama/Native/SafeLLamaGrammarHandle.cs

```diff
@@ -97,11 +88,11 @@ public static SafeLLamaGrammarHandle Create(IReadOnlyList<GrammarRule> rules, ul
         /// <exception cref="RuntimeError"></exception>
         public static unsafe SafeLLamaGrammarHandle Create(LLamaGrammarElement** rules, ulong nrules, ulong start_rule_index)
         {
-            var grammar_ptr = NativeApi.llama_grammar_init(rules, nrules, start_rule_index);
-            if (grammar_ptr == IntPtr.Zero)
+            var grammar = NativeApi.llama_grammar_init(rules, nrules, start_rule_index);
+            if (grammar == null)
```

Comment:
I might be a bit picky, but will it be better to use `is` to check the null? [Here](https://medium.com/@erickgallani/the-difference-between-null-check-with-is-instead-of-in-c-ed3d955a639e#id_token=eyJhbGciOiJSUzI1NiIsImtpZCI6ImUxYjkzYzY0MDE0NGI4NGJkMDViZjI5NmQ2NzI2MmI2YmM2MWE0ODciLCJ0eXAiOiJKV1QifQ.eyJpc3MiOiJodHRwczovL2FjY291bnRzLmdvb2dsZS5jb20iLCJhenAiOiIyMTYyOTYwMzU4MzQtazFrNnFlMDYwczJ0cDJhMmphbTRsamRjbXMwMHN0dGcuYXBwcy5nb29nbGV1c2VyY29udGVudC5jb20iLCJhdWQiOiIyMTYyOTYwMzU4MzQtazFrNnFlMDYwczJ0cDJhMmphbTRsamRjbXMwMHN0dGcuYXBwcy5nb29nbGV1c2VyY29udGVudC5jb20iLCJzdWIiOiIxMTUwNzU5Mjk3MzIyMDQ1NzYzMjgiLCJlbWFpbCI6ImFzYWt1c2FyaW5uZUBnbWFpbC5jb20iLCJlbWFpbF92ZXJpZmllZCI6dHJ1ZSwibmJmIjoxNzE0MTIyOTQ4LCJuYW1lIjoiUmlubmUiLCJwaWN0dXJlIjoiaHR0cHM6Ly9saDMuZ29vZ2xldXNlcmNvbnRlbnQuY29tL2EvQUNnOG9jSTdzYlltRnR5YmhTT2g2R0FsLUdyTVpJNzVOWFdmRmphTEdfb3l0RVRTaW5BTXlnWT1zOTYtYyIsImdpdmVuX25hbWUiOiJSaW5uZSIsImlhdCI6MTcxNDEyMzI0OCwiZXhwIjoxNzE0MTI2ODQ4LCJqdGkiOiI5MzMxZmI2Mzk3Njc3MmE0Y2IwNTdkYTdiMWYzNTRhZDBkOGY4NGMyIn0.XYGLCsXKsy07ZlLu68swEgZ6U3r8m3HH3lfGjgVoXE6LWus3Tj7I1R_aFY_UNG7G0w336-JkODz_z3J4LYZyv7S2pze69KMxOeBEGzWTLT8Jx4L8ajLUWsxPM63RfZS7wf4NAXbQ4O12mXt67Av3D2hvTf4SO31b1wUCES1Kp2BXJAPI69SoG8sd8p1WPEkoGfNBoLrokLTaRBB6x6J-fXzSKWj1PrcYC6VGF2NhD3yyZ7QIAm1Tfxvwp5p_fyAH14EJTk_yPeBbZkHrC_oW_KojPMv2_oTZ2bCTZdLMhWkAwBZrstHTUsdsuydv2ukfjn9p8GUBSV6Y9n7hAnMdbg) is an article to tell the difference between them.

Reply:
Changed to use an `is` check